### PR TITLE
Adding skonsole workflow to generate PR descriptions.

### PR DESCRIPTION
**Pull request by mkarle**
### Motivation and Context

This adds a PR comment-triggered workflow that uses the action [mkarle/skonsole-generate-pr-description](https://github.com/mkarle/skonsole-generate-pr-description) to run skonsole on the diff of the PR.  That action uses [mkarle/pr-update](https://github.com/mkarle/pr-update) to update the PR with a title, a summary, and a list of changes.  It can update as a prefix, suffix, or entirely replace what's there.

### Description

To use, create a comment on a PR with:
`/sk generate-pr-description <suffix(default) | prefix | replace>`
Since it's triggered from a pull request comment, it only runs if it's in the default branch of the repo. [See this PR for an example of it working.](https://github.com/mkarle/skonsole/pull/4)
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by lemillermicrosoft — APPROVED** [review]

**Comment by lemillermicrosoft** [comment]
@microsoft/octo-semantickernel-pr-apps FYI

File: .github/workflows/generate-pr-description.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Generate PR description
+        uses: mkarle/skonsole-generate-pr-description@v1
+        with:
+          pull-request-number: ${{ github.event.issue.number }}
+          pull-request-diff-url: ${{ github.event.issue.pull_request.diff_url }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          update-type:  ${{ contains(github.event.comment.body, 'replace') && 'replace' || (contains(github.event.comment.body, 'prefix') && 'prefix' || 'suffix') }}
+        env: # Set Azure credentials secret as an input
+          AZURE_OPENAI_CHAT_DEPLOYMENT_NAME: ${{ vars.AZUREOPENAI__CHAT__DEPLOYMENTNAME }}
```
Comment (lemillermicrosoft) [inline .github/workflows/generate-pr-description.yml:34]: Perfect for now -- we may want a separate variable name if we want toggle it without impacting other actions.

**Comment by hario90** [comment]
This is really cool! It would be nice to see the prefix and/or suffix options in action as well.

File: .github/workflows/generate-pr-description.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: pending
+
+      - name: Generate PR description
+        uses: mkarle/skonsole-generate-pr-description@v1
+        with:
+          pull-request-number: ${{ github.event.issue.number }}
+          pull-request-diff-url: ${{ github.event.issue.pull_request.diff_url }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          update-type:  ${{ contains(github.event.comment.body, 'replace') && 'replace' || (contains(github.event.comment.body, 'prefix') && 'prefix' || 'suffix') }}
```
[the remark this answers is not in the document]
Comment (hario90) [reply .github/workflows/generate-pr-description.yml:32]: Oh nevermind, makes sense now.

[the remark this answers is not in the document]
Comment (mkarle) [reply .github/workflows/generate-pr-description.yml:32]: Yeah, github action tertiary operators are ugly. Basically `booleanValue && resultIfTrue || resultIfFalse` is equivalent to `booleanValue ? resultIfTrue : resultIfFalse`

File: .github/workflows/generate-pr-description.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  issue_comment:
+    types: [created]
+
+jobs:
+  generate-pr-description:
+    permissions:
+      pull-requests: write
+      statuses: write
+    runs-on: ubuntu-latest
+    if: github.event.issue.pull_request && contains(github.event.comment.body, '/sk generate-pr-description')
```
[the remark this answers is not in the document]
Comment (mkarle) [reply .github/workflows/generate-pr-description.yml:13]: That's a good idea. I'll make that next.

**Comment by mkarle** [comment]
> This is really cool! It would be nice to see the prefix and/or suffix options in action as well.

In the [PR I linked in the description](https://github.com/mkarle/skonsole/pull/4) I used replace, and it defaults to suffix. There's an example of prefix [in this one](https://github.com/mkarle/skonsole/pull/1)

**Review by hario90 — APPROVED** [review]

**Pull request merged**